src/Catalog/Helpers/Utils.cs

```diff
@@ -44,38 +51,94 @@ public static string GetResource(string resName)
 
             return _resourceStringCache[resName];
         }
-
-        public static IGraph CreateNuspecGraph(XDocument nuspec, string baseAddress)
+        
+        public static IGraph CreateNuspecGraph(XDocument nuspec, string baseAddress, bool normalizeXml = false)
         {
-            nuspec = NormalizeNuspecNamespace(nuspec);
-
-            XslCompiledTransform transform = CreateTransform("xslt.nuspec.xslt");
-
             XsltArgumentList arguments = new XsltArgumentList();
-            arguments.AddParam("base", "", baseAddress + "packages/");
+            arguments.AddParam("base", "", baseAddress);
             arguments.AddParam("extension", "", ".json");
 
             arguments.AddExtensionObject("urn:helper", new XsltHelper());
 
-            XDocument rdfxml = new XDocument();
-            using (XmlWriter writer = rdfxml.CreateWriter())
+            nuspec = SafeXmlTransform(nuspec.CreateReader(), XslTransformNormalizeNuSpecNamespaceCache.Value);
+            var rdfxml = SafeXmlTransform(nuspec.CreateReader(), XslTransformNuSpecCache.Value, arguments);
+
+            var doc = SafeCreateXmlDocument(rdfxml.CreateReader());
+            if (normalizeXml)
             {
-                transform.Transform(nuspec.CreateReader(), arguments, writer);
+                NormalizeXml(doc);
             }
 
             RdfXmlParser rdfXmlParser = new RdfXmlParser();
-            XmlDocument doc = new XmlDocument();
-            doc.Load(rdfxml.CreateReader());
             IGraph graph = new Graph();
             rdfXmlParser.Load(graph, doc);
 
             return graph;
         }
 
-        static XslCompiledTransform CreateTransform(string name)
+        private static void NormalizeXml(XmlNode xmlNode)
+        {
+            if (xmlNode.Attributes != null)
+            {
+                foreach (XmlAttribute attribute in xmlNode.Attributes)
+                {
+                    attribute.Value = attribute.Value.Normalize(NormalizationForm.FormC);
+                }
+            }
+
+            if (xmlNode.Value != null)
+            {
+                xmlNode.Value = xmlNode.Value.Normalize(NormalizationForm.FormC);
+                return;
+            }
+
+            foreach (XmlNode childNode in xmlNode.ChildNodes)
+            {
+                NormalizeXml(childNode);
+            }
+        }
+
+        internal static XmlDocument SafeCreateXmlDocument(XmlReader reader = null)
+        {
+            // CodeAnalysis / XmlDocument: set the resolver to null or instance
```

Comment:
Is there an error code associated with the CodeAnalysis here? Usually it's like AA000 or something. Could we add that to the messages here?